grunt_tasks/build.js

```diff
@@ -14,301 +14,98 @@
+            watch: _.extend({}, isDev ? webpackDevConfig : webpackProdConfig, {
+                watch: true,
+                keepalive: true
+            })
         },
-        module: {
-            loaders: [
-                { // ES2015
-                    test: /\.js$/,
-                    loader: 'babel-loader',
-                    exclude: [paths.node_modules]
-                },
-                { // Stylus
-                    test: /\.styl$/,
-                    loaders: [
-                        ExtractTextPlugin.extract('style-loader'),
-                        'css-loader',
-                        {
-                            loader: 'stylus-loader',
-                            query: {
-                                'resolve url': true
-                            }
-                        }
-                    ]
-                },
-                { // CSS
-                    test: /\.css$/,
-                    loaders: [
-                        ExtractTextPlugin.extract('style-loader'),
-                        'css-loader'
-                    ]
-                },
-                { // Jade
-                    test: /\.jade$/,
-                    loader: 'jade-loader',
-                    query: {
-                        doctype: 'html' // see @girder/pull/1469
-                    }
-                },
-                { // PNG, JPEG
-                    test: /\.(png|jpg)$/,
-                    loaders: [urlLoader(), fileLoader()]
-                },
-                { // WOFF
-                    test: /\.woff(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [urlLoader({ mimetype: 'application/font-woff' }), fileLoader()]
-                },
-                { // WOFF2
-                    test: /\.woff2(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [urlLoader({ mimetype: 'application/font-woff2' }), fileLoader()]
-                },
-                { // TTF
-                    test: /\.ttf(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [urlLoader({ mimetype: 'application/octet-stream' }), fileLoader()]
-                },
-                { // EOT
-                    test: /\.eot(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [fileLoader()]
-                },
-                { // SVG
-                    test: /\.svg(\?v=\d+\.\d+\.\d+)?$/,
-                    loaders: [urlLoader({ mimetype: 'image/svg+xml' }), fileLoader()]
+        // The grunt-contrib-watch task below is one option for dev, as described here:
+        // https://github.com/webpack/webpack-with-common-libs/blob/master/Gruntfile.js
+        // but it could also be done using the webpack dev server.
+        // It is however A LOT SLOWER than using the built-in watch options in grunt-webpack
+        watch: {
+            app: {
+                files: getWatchFiles(paths.web_src),
+                tasks: ['warnWatch', isDev ? 'webpack:dev' : 'webpack:prod'],
```

Comment:
It's good to hear. I did put that warning because there still is *one* remaining `grunt-contrib-watch` task at play here, the one used to watch the Sphinx doc (see `sphinx.hs`). Now the issue is that you can't use both at the same time, because both `grunt-contrib-watch` or `grunt-webpack` (in watch mode) would try take control (or `webpack-dev-server` for that matter). It was working before because `grunt-contrib-watch` would watch everything (doc files, javascript, jade, etc) but this is neither viable nor desirable now that Webpack is in control of the build process. My gut feeling is that "watching" the build process is a more common case than watching the documentation process, so I would probably favor `npm run watch` triggering the `webpack:watch` task. We can chew on that.